### ISSUE TITLE
Render traffic lights, stop signs (and similar) in driving mode

### ISSUE THREAD

**Issue by euf** [organicmaps/organicmaps#3987]
Apple Maps and Google Maps do a good job of showing traffic lights and stop signs on the route while driving.

![D739FD40-F89A-4432-98D1-A9C0EC5BBF2F](https://user-images.githubusercontent.com/11754264/204104845-57d80597-2848-429c-a733-e6aa2d09c3b1.jpeg)

If I understand OM styling correctly, these POIs could be made to be shown only in the `vehicle` theme. Anyway I'm not sure if it's possible to control their Z index so they're drawn on top of the route blue line. I hope someone with a better knowledge of the code could provide some insight on that. 

In the meantime, I'll try to draw some SVGs for the corresponding POIs:
- highway=traffic_signals
- highway=stop
- highway=give_way
- traffic_calming=yes,bump,mini_bumps,hump,table,cushion,dip 
- highway=crossing with crossing=uncontrolled, crossing=marked, crossing=zebra, crossing=unmarked

Discussion on the importance of these features for the driver as well as further suggestions are welcome.

**Comment by biodranik** [comment]
@vng does it make sense to add these features to the normal POI section? Or we can add them only to the routing graph and extract them from there? What if we later decide to allow adding these features via Editor? (although it's a bit complex because the direction and the lane of these features is important).

**Comment by aronkvh** [comment]
It would make sense to add them to the regular view as they are usually easy to spot from further so can act as a point of reference.

**Comment by biodranik** [comment]
> It would make sense to add them to the regular view

I don't think that it is a good idea. These signs are needed only for drivers. No need to show them to pedestrians/tourists. It may make some sense when we implement a special "driving" map style.

**Comment by jmarchon1** [comment]
I'd love this!

**Comment by euf** [comment]
> What if we later decide to allow adding these features via Editor?

I may speak from my own experience, but for the reasons you described

> the direction and the lane of these features is important

It seems hard to think of a good mobile UX for this task. Personally I tried to overcome this in Osmand by setting up shortcuts for leaving predefined OSM notes (here's a bump, or pedestrian crossing, or 30 km/h speed limit) to minimize my driving distraction. Then I would close them myself on the desktop or leave them for future armchair mappers.

My point is I don't map these features on the go when I'm walking (cause as a pedestrian I simply don't care) and when I'm driver (that would be dangerous). 

Providing a one, maximum two click interface for leaving notes on speed limits (2gis or Yandex had it when tapping on a speed indicator) would be nice. Trying to make traffic lights and stop signs properly editable on mobile might be both hard and unnecessary.

P.S. I just checked EveryDoor, and even it (with all the node-snapping-way magic) fails to provide a proper UX for choosing a highway=stop direction.

**Comment by muralito** [comment]
> > It would make sense to add them to the regular view
> 
> I don't think that it is a good idea. These signs are needed only for drivers. No need to show them to pedestrians/tourists. It may make some sense when we implement a special "driving" map style.

When a tourist stops you and asks for directions, it is useful to reply them "walk until the next traffic lights, and turn left...", but I suppose that if the tourist use OM they shouldn't be asking me :-) . It may seem that guiding them using "meters" in the directions is enough, but maybe saying "walk three blocks and turn left" is better than "walk 270 meters and turn left". Probably the right answer is culture dependent, or language dependent. IMO "three blocks" seems more close, more "human", than "270 meters". Probably is also age dependent, i have been using gps navigation only in the last 10 years, previous to that you had to open the window and ask someone how to get "there"...

**Comment by biodranik** [comment]
Does it make sense to announce bumps and/or some other important road signs? Maybe depending on the current speed.

**Comment by matheusgomesms** [comment]
> Does it make sense to announce bumps and/or some other important road signs? Maybe depending on the current speed.

Funny I was thinking about this issue recently! Maybe as an option, like Always/Never/Only when high speed, but if this is too hard maybe no announcement or only when driving a bit fast (let's say above 40-50 km/h?). Otherwise the TTS would be very annoying (especially in USA, where some places have STOP signs in every intersection).
